qiita_db/metadata_template.py

```diff
@@ -2228,20 +2229,32 @@ def load_template_to_dataframe(fn, strip_whitespace=True):
     |             longitude |        float |
     +-----------------------+--------------+
     """
+    # Load in file lines
+    holdfile = None
+    if isinstance(fn, (str, unicode)):
+        # filepath passed
+        with open(fn) as f:
+            holdfile = f.readlines()
+    else:
+        # open file passed
+        holdfile = fn.readlines()
+    if len(holdfile) == 0:
+        raise ValueError('Empty file passed!')
 
-    # First, strip all values from the cells in the input file, if requested
+    # Strip all values in the cells in the input file, if requested
     if strip_whitespace:
-        fd, fp = mkstemp()
-        close(fd)
-
-        with open_file(fn, 'U') as input_f, open(fp, 'w') as new_f:
-            for line in input_f:
-                line_elements = [x.strip()
-                                 for x in line.rstrip('\n').split('\t')]
-                new_f.write('\t'.join(line_elements) + '\n')
-
-        fn = fp
-
+        for pos, line in enumerate(holdfile):
+            holdfile[pos] = '\t'.join(d.strip(' ') for d in line.split('\t'))
```

Comment:
This is only stripping spaces, and not other whitespace characters. Although tab and newline have special meaning in the TSV file and will probably not be found, there are other whitespace characters that might be present in the file that should also be stripped (such as space, tab, linefeed, return, formfeed, or vertical tab). I recommend leaving the argument to `strip` blank.